thrust/testing/cuda/remove.cu

```diff
@@ -329,23 +329,18 @@ void TestRemoveCudaStreams()
   using Vector = thrust::device_vector<int>;
   using T      = Vector::value_type;
 
-  Vector data(5);
-  data[0] = 1;
-  data[1] = 2;
-  data[2] = 1;
-  data[3] = 3;
-  data[4] = 2;
+  Vector data{1, 2, 1, 3, 2};
 
   cudaStream_t s;
   cudaStreamCreate(&s);
 
   Vector::iterator end = thrust::remove(thrust::cuda::par.on(s), data.begin(), data.end(), (T) 2);
 
   ASSERT_EQUAL(end - data.begin(), 3);
+  data.resize(end - data.begin());
```

Comment:
Suggestion: the canonical action after `remove` is to erase the elements from the returned iterator to the end:
```suggestion
  data.erase(end, data.end());
```
Applies to a few more places below.